### ISSUE TITLE
tira as que estao em duvida

### ISSUE THREAD

**Issue by VascoLopo** [VascoLopo/ajuda#2]
import pandas as pd
# Create an empty DataFrame with the desired columns
df = pd.DataFrame(columns=['ID', 'Embeddings', 'Labels', 'POS Tags', 'Parse Trees'])

# Iterate over the embeddings, labels, POS tags, and parse trees
for i in range(len(embeddings)):
    # Modify the 'labels' list so that any case where labels are both 'Populism' and 'Non populism' is counted only as 'Populism'
    
    labels[i] = ['Populism', 'Populism'] if set(labels[i]) == set(['Populism', 'Non populism']) or set(labels[i]) == set( ['Non populism', 'Populism'])  else labels[i]

    # Create a new row of data for each item
    row_data = {'ID': i, 'Embeddings': embeddings[i], 'Labels': labels[i], 'POS Tags': pos_tags[i], 'Parse Trees': parse_trees[i]}
    # Append the row to the DataFrame
    df = df.append(row_data, ignore_index=True)

# Set the display option to show all columns
pd.set_option('display.max_columns', None)

# Print the DataFrame
print(df.head(10))

sizes = set()
for i in range(len(embeddings)):
    for embedding in embeddings[i]:
        size = len(embedding)
        sizes.add(size)
print(sizes)

from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, precision_score, recall_score, f1_score
import numpy as np

# Convert the modified 'labels' list to a numpy array
labels_array = np.array(labels)

# Convert the embeddings list to a 2D numpy array
embeddings_array = np.array(embeddings)

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(embeddings_array, labels_array, test_size=0.2, random_state=42)

# Create a logistic regression model
model = LogisticRegression(random_state=42)

# Fit the model to the training data
model.fit(X_train, y_train)

# Make predictions on the testing data
y_pred = model.predict(X_test)

# Compute evaluation metrics
accuracy = accuracy_score(y_test, y_pred)
conf_matrix = confusion_matrix(y_test, y_pred)
precision = precision_score(y_test, y_pred, average='macro')
recall = recall_score(y_test, y_pred, average='macro')
f1 = f1_score(y_test, y_pred, average='macro')

# Print the evaluation metrics
print("Accuracy:", accuracy)
print("Confusion matrix:\n", conf_matrix)
print("Precision:", precision)
print("Recall:", recall)
print("F1-score:", f1)
